### PR TITLE
Activity Table date label formatting

### PR DESCRIPTION
**Pull request by BrianBelhumeur**
# Description
This PR builds upon @blueridger's work implementing the date interval selection. These changes include:
- Date string formatting as per issue #2260
-- Show start year only when different from end year
-- Show "Today" instead of current or future date
- Default to "Last Week's Activity" time interval on page load
- Eliminated a visible repaint by refactoring interval selection out of `useEffect` and into a small helper function

# Test Plan
Tested manually, viewing date formats when switching between each date interval

## Before
https://www.loom.com/share/4d5b8af1b2c34bb7824e4f3ef8841345

## After
https://www.loom.com/share/e5e20f03e95d400ca846d08dde893ced

### CODE REVIEW

File: src/ui/components/ExplorerHome/ExplorerHome.js [hunk 1]
```diff
@@ -354,22 +349,51 @@ export const ExplorerHome = ({
...
+
+      return someDate >= today;
+    };
+
+    const currentYear = new Date().getFullYear();
+    const startDate = new Date(interval.startTimeMs);
+    const endDate = new Date(interval.endTimeMs);
+
+    const startTime =
+      startDate.getFullYear() === currentYear
```
Comment (blueridger) [inline src/ui/components/ExplorerHome/ExplorerHome.js:386]: Isnt this supposed to compare the startDate year to the _endDate year_, not the current year.

File: src/ui/components/ExplorerHome/ExplorerHome.js [hunk 2]
```diff
@@ -354,22 +349,51 @@ export const ExplorerHome = ({
...
+        minute: undefined,
+        year: undefined,
+      });
+
+    const isTodayOrLater = (someDate) => {
+      const today = new Date();
+      today.setHours(0);
+      today.setMinutes(0);
+      today.setSeconds(0);
+      today.setMilliseconds(0);
```
Comment (blueridger) [inline src/ui/components/ExplorerHome/ExplorerHome.js:376]: Fun fact, these four lines can be replaced with `today.setHours(0, 0, 0, 0)`

Comment (BrianBelhumeur) [reply src/ui/components/ExplorerHome/ExplorerHome.js:376]: Oooh that is fun, thank you!

File: src/ui/components/ExplorerHome/ExplorerHome.js (hunk 1 above)
Comment (BrianBelhumeur) [reply src/ui/components/ExplorerHome/ExplorerHome.js:386]: Ah good point. In my head I was assuming end date year would always be the current year, but I can see how that would not necessarily be the case. Good catch, and thank you.

**Review by KuraFire — APPROVED** [review]
Ya lgtm! 👍🏼

**Review by blueridger — APPROVED** [review]

**Pull request merged**